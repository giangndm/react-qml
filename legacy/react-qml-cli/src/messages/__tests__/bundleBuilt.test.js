import message from '../bundleBuilt';

const statsWithWithoutWarnings = {
  toJson: () => ({
    time: 1000.0203,
    warnings: [],
  }),
  hasWarnings: () => false,
};

const statsWithWithWarnings = {
  toJson: () => ({
    children: [
      {
        time: 1000.21312,
      },
      {
        time: 2300.123,
      },
    ],
    warnings: ['warn1', 'warn2'],
  }),
  hasWarnings: () => true,
};

describe('message bundleBuild for all platforms', () => {
  it('should match the snapshot without warnings', () => {
    expect(
      message({ stats: statsWithWithoutWarnings, platform: 'all' })
    ).toMatchSnapshot();
  });

  it('should match the snapshot with warnings', () => {
    expect(
      message({ stats: statsWithWithWarnings, platform: 'all' })
    ).toMatchSnapshot();
  });
});
