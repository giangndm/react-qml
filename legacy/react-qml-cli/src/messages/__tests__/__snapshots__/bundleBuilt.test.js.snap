// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`message bundleBuild for all platforms should match the snapshot with warnings 1`] = `
"Built with warnings in 2.30s!

warn1

warn2

You can now run the app on your device
"
`;

exports[`message bundleBuild for all platforms should match the snapshot without warnings 1`] = `
"Built successfully in 1.00s!

You can now run the app on your device
"
`;
